### PR TITLE
fix: enable tracking number creation without verification

### PR DESCRIPTION
**Pull request by gca-axelor**
RM#82115

### CODE REVIEW

File: packages/apps/stock/src/screens/supplierArrivals/SupplierArrivalLineDetailScreen.js [hunk 1]
```diff
@@ -234,6 +244,15 @@ const SupplierArrivalLineDetailScreen = ({route, navigation}) => {
...
         />
+        <TouchableOpacity
+          style={styles.trackingNumberContainer}
+          onPress={handleAddTrackingNumber}
+          disabled={readonly}>
+          <Text writingType="important" style={styles.trackingNumberText}>
+            {I18n.t('Stock_AddTrackingNumber')}
+          </Text>
+          <Icon name="plus-lg" size={24} style={styles.action} />
+        </TouchableOpacity>
```
Comment (lme-axelor) [inline packages/apps/stock/src/screens/supplierArrivals/SupplierArrivalLineDetailScreen.js:255]: Create component for that
Also, it should be display only on conditions : product.trackingNumberConfiguration != null and trackingNumber == null
User should be also able to select an existing tracking number

File: changelogs/unreleased/82115.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "title": "SupplierArrivalLineDetailScreen: possibility to add tracking number without verification",
```
Comment (lme-axelor) [inline changelogs/unreleased/82115.json:2]: Supplier arrival: add possibility to register a tracking number when the line is incomplete.

File: packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelection/SupplierArrivalTrackingNumberSelection.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+const SupplierArrivalTrackingNumberSelection = ({
+  onSelectTrackingNumber,
+  onAddTrackingNumber,
+  product,
+  readonly,
+}) => {
+  const Colors = useThemeColor();
+  const I18n = useTranslator();
+
+  const trackingScanKey = 'tracking_supplier-arrival-select';
```
Comment (vhu-axelor) [inline packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelection/SupplierArrivalTrackingNumberSelection.js:34]: In general we put key outside of the component

File: packages/apps/stock/src/screens/supplierArrivals/SupplierArrivalLineDetailScreen.js [hunk 4]
```diff
@@ -244,6 +275,15 @@ const styles = StyleSheet.create({
...
   },
+  trackingNumberContainer: {
+    flexDirection: 'row',
+    alignSelf: 'flex-start',
+    marginHorizontal: 24,
+    marginTop: 10,
+  },
+  action: {
+    marginLeft: 10,
+  },
```
Comment (vhu-axelor) [inline packages/apps/stock/src/screens/supplierArrivals/SupplierArrivalLineDetailScreen.js:286]: You can remove these styles

File: packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelection/SupplierArrivalTrackingNumberSelection.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+const styles = StyleSheet.create({
+  trackingNumberContainer: {
+    flexDirection: 'row',
+    alignSelf: 'flex-start',
+    marginHorizontal: 24,
+    marginTop: 10,
+  },
+  text_secondary: {
```
Comment (vhu-axelor) [inline packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelection/SupplierArrivalTrackingNumberSelection.js:71]: textSecondary or text

File: packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelect/SupplierArrivalTrackingNumberSelect.js
Comment (vhu-axelor) [inline packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelect/SupplierArrivalTrackingNumberSelect.js:1]: Remove View and return null if canCreate is false

File: packages/apps/stock/src/features/trackingNumberSlice.js [hunk 6]
```diff
@@ -81,6 +82,25 @@ export const updateSupplierTrackingNumber = createAsyncThunk(
   },
 );
 
+export const updateStockMoveLineTrackingNumber = createAsyncThunk(
+  'trackingNumber/updateStockMoveLineTrackingNumber',
+  async function (data, {getState, dispatch}) {
+    return handlerApiCall({
+      fetchFunction: _updateStockMoveLineTrackingNumber,
+      data,
+      action: 'Stock_SliceAction_UpdateSupplierArrivalLineWithTrackingNumber',
```
Comment (vhu-axelor) [inline packages/apps/stock/src/features/trackingNumberSlice.js:91]: Change trads

File: packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelect/SupplierArrivalTrackingNumberSelect.js [hunk 7]
```diff
@@ -67,48 +57,17 @@ const SupplierArrivalTrackingNumberSelect = ({
     }
   };
 
+  if (!canCreate) {
+    return null;
+  }
```
Comment (lme-axelor) [inline packages/apps/stock/src/components/templates/supplierArrival/SupplierArrivalTrackingNumberSelect/SupplierArrivalTrackingNumberSelect.js:62]: It's functionnally false, canCreate permission should show/hide only the creation part of the component, not the selection part. 
Also why having two components SuppleirArrivalTrackingNumberSelect & SupplierArrivalTrackingNumberSelection which will have the same navigation on creation, permission check,...
try to refactor to have only one component, the only difference should be the onSelectTrackingNumber which can be given in props for both screens that use it.

**Pull request merged**